common/hooks/useValidation.js

```diff
@@ -0,0 +1,10 @@
+import { useState } from 'react';
+
+const useValidation = (initIsValid = true, initShowValidity = false) => {
+  const [isValid, setIsValid] = useState(initIsValid);
+  const [showValidity, setShowValidity] = useState(initShowValidity);
+
+  return { isValid, setIsValid, showValidity, setShowValidity };
+};
+
+export default useValidation;
```

Comment:
To make sure I am clear here - we're using the hook to pass data from the input back to the parent component?

Reply:
Trying to think how this abstraction would scale? If we had a form with 5 fields, we would create a validation for each and then check each on the submission of the form?

Reply:
>  using the hook to pass data from the input back to the parent component?

yes (because more often than not an input would need, as you say, to have its validity (re-)checked at the point the form was submitted.

And yes, I was envisaging a `useValidation` hook for any input that required validation that the form had access to.